### PR TITLE
Use ruff instead of black+flake8

### LINKED ISSUE #1670: Consider migration to ruff
It may make sense to use `ruff` instead of `black` + `flake8`. Figure out what needs to be done and whether there are any serious challenges to implement this step.

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Resolves #1670 
Switched to ruff. It works great, had to make some modifications though.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [x] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by PPsyrius** [comment]
From a quick glance, it seems like you'll need to set a separate [max-string-length](https://docs.astral.sh/ruff/settings/#flake8-errmsg-max-string-length) to 99 again, too, in the [`pyproject.toml`](https://github.com/vacanza/python-holidays/blob/beta/pyproject.toml).

```toml
[tool.ruff.lint.flake8-errmsg]
max-string-length = 99
```

**Review by arkid15r — COMMENTED** [review]
I've added `flake8-errmsg` suggestion (even though it doesn't seem to make any difference for now). The E203 one is still in [preview](https://docs.astral.sh/ruff/rules/whitespace-before-punctuation/), I guess it can wait.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -63,6 +47,12 @@ repos:
           - rstcheck
           - sphinx
 
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.13
```
Comment (KJhellico) [inline .pre-commit-config.yaml:51]:
```suggestion
    rev: v0.1.14
```
And maybe place this block above, in place of `black`?

Comment (arkid15r) [reply .pre-commit-config.yaml]: The ordering wasn't right, thank you!

File: tests/countries/test_thailand.py [hunk 2]
```diff
@@ -159,10 +162,7 @@ def test_2022_all(self):
             ("2022-09-28", "วันพระราชทานธงชาติไทย"),
             (
                 "2022-10-13",
-                (
-                    "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร "
-                    "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร"
-                ),
+                ("วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร " "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร"),
```
Comment (KJhellico) [inline tests/countries/test_thailand.py:165]:
```suggestion
                "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร",
```

**Comment by KJhellico** [comment]
LGTM! let's ruff it! 🚀

**Review by arkid15r — COMMENTED** [review]
All right, I've fixed those " " cases!

Thanks for the thorough review! 👍

**Review by KJhellico — APPROVED** [review]

**Pull request merged**